File: SampleAppObjC/SampleAppObjC/PushPackageTestingViewController.m
//
//  PushPackageTestingViewController.m
//  SampleAppObjC
//
//  Created by Steven Gerhard on 7/16/19.
//  Copyright © 2019 LaunchKey. All rights reserved.
//

#import "PushPackageTestingViewController.h"
#import <QuartzCore/QuartzCore.h>

@interface PushPackageTestingViewController ()
@property (weak, nonatomic) IBOutlet UILabel *labelTestPushPackage;
@property (strong, nonatomic) IBOutlet UITextField *pushPackageTextField;
@end

@implementation PushPackageTestingViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    if (@available(iOS 11.0, *)) {
        _pushPackageTextField.attributedPlaceholder = [[NSAttributedString alloc] initWithString:@"Push Package" attributes:@{NSForegroundColorAttributeName:[UIColor colorNamed:@"placeholderTextColor"]}];
        _labelTestPushPackage.textColor = [UIColor colorNamed:@"viewTextColor"];

    } else {
        _pushPackageTextField.attributedPlaceholder = [[NSAttributedString alloc] initWithString:@"Push Package" attributes:@{NSForegroundColorAttributeName:[UIColor lightGrayColor]}];
        _labelTestPushPackage.textColor = [UIColor blackColor];
    }
}

- (IBAction)pressedSubmitButton:(UIButton *)sender {
    NSString *text = [[self pushPackageTextField] text];
    if (text != nil) {
        if (text.length > 0) {
            [[LKCAuthenticatorManager sharedClient] handleThirdPartyPushPackage:text];
        }
    }
        
}

@end

File: SampleAppSwift/SampleAppSwift/PushPackageTestingViewController.swift
//
//  PushPackageTestingViewController.swift
//  SampleAppSwift
//
//  Created by Steven Gerhard on 7/15/19.
//  Copyright © 2019 LaunchKey. All rights reserved.
//

import Foundation
import UIKit

class PushPackageTestingViewController : UIViewController {
    @IBOutlet var pushPackageTextField: UITextField!
    @IBOutlet weak var labelTestPushPackage: UILabel!
    
    override func viewDidLoad() {
        if #available(iOS 11.0, *) {
            pushPackageTextField.attributedPlaceholder = NSAttributedString(string: "Push Package", attributes: [NSAttributedString.Key.foregroundColor : UIColor(named: "placeholderTextColor")!])
            labelTestPushPackage.textColor = UIColor(named:"viewTextColor")
        } else {
            pushPackageTextField.attributedPlaceholder = NSAttributedString(string: "Push Package", attributes: [NSAttributedString.Key.foregroundColor : UIColor.lightGray])
            labelTestPushPackage.textColor = UIColor.black
        }
    }
    
    @IBAction func pressedSubmit(_ sender: UIButton) {
        if let payload = pushPackageTextField.text, payload.count > 0 {
            LKCAuthenticatorManager.sharedClient().handleThirdPartyPushPackage(payload)
        }
    }
}
